components/ILIAS/Mail/classes/class.ilObjMailGUI.php

```diff
@@ -578,6 +581,23 @@ protected function getExternalSettingsForm(): ilPropertyFormGUI
         return $form;
     }
 
+    private function buildSignaturePlaceholderInput(Signature $signature): ilManualPlaceholderInputGUI
+    {
+        $placeholder_input = new ilManualPlaceholderInputGUI(
+            $this->lng->txt('mail_form_placeholders_label'),
+            $signature->getSettingsKeyword()
+        );
+        $placeholder = $this->mail_signature_service->getPlaceholder();
+        while ($placeholder->getNext()) {
+            if ($signature->supports($placeholder)) {
+                $placeholder_input->addPlaceholder($placeholder->getId(), $placeholder->getLabel());
+            }
+            $placeholder = $placeholder->getNext();
+        }
+        $placeholder_input->setDisabled(!$this->isEditingAllowed());
+        return $placeholder_input;
```

Comment:
Do we need a "do/while" loop instead? What if the first `$placeholder` does not provide a next element here?

```suggestion
        $placeholder_input = new ilManualPlaceholderInputGUI(
            $this->lng->txt('mail_form_placeholders_label'),
            $signature->getSettingsKeyword()
        );
        $placeholder_input->setDisabled(!$this->isEditingAllowed());

        $placeholder = $this->mail_signature_service->getPlaceholder();
        while ($placeholder->getNext()) {
            if ($signature->supports($placeholder)) {
                $placeholder_input->addPlaceholder($placeholder->getId(), $placeholder->getLabel());
            }
            $placeholder = $placeholder->getNext();
        }

        return $placeholder_input;
```